Automattic/wp-calypso
# Checkout: Collect contact information for all purchases

Currently when a customer checks out with a domain in their cart, the checkout process will ask for contact information which is used for the domain registration. This PR adds an AB test to begin asking customers for contact information on every checkout instead (although it also includes the option to enter separate contact information for a domain, if desired).

Our hope is to simplify the checkout process with fewer screens, thereby losing fewer customers in checkout.

**This PR:**

- Adds a `collectBillingInformation` AB test
- Adds the "Billing Information" form to checkout
- Adds checkboxes for domain privacy and (optional) alternate domain contact information, if a domain is present in the cart
- Shows a second contact information form, if the alternate domain contact information option is selected
- Adds/removes domain privacy from the cart with the corresponding checkbox

#### Testing instructions

* Apply D19756-code to your sandbox and points it to `public-api.wordpress.com`.
* Make sure you're assigned to the `test` variation of `collectBillingInformation` A/B test.
* Try to purchase a paid plan without domains. You will see the billing infomartation forms like below:
  <img width="1034" alt="2018-10-23 4 35 40" src="https://user-images.githubusercontent.com/212034/47314633-2c763300-d67d-11e8-919c-f59f222563eb.png">
* Once you complete the transaction, the form would be pre-filled next time.
* Try to buy a paid plan _**with**_ a domain. The form will look like the following:
  <img width="1027" alt="2018-10-23 4 38 36" src="https://user-images.githubusercontent.com/212034/47314758-9393e780-d67d-11e8-8f47-d93ea91b588f.png">
* An additional domain contact form shows up when _"Use this as my domain contact information"_ is unchecked.
